### PR TITLE
Enforce offline volume expansion

### PR DESCRIPTION
**Pull request by gnufied**
Also Volume struct does not need mutex in it. No code is using it afaict.

/sig storage

### CODE REVIEW

File: mock/service/controller.go [hunk 1]
```diff
@@ -227,6 +227,11 @@ func (s *service) ControllerPublishVolume(
 	v.VolumeContext[ReadOnlyKey] = roVal
 	s.vols[i] = v
 
+	if volInfo, ok := MockVolumes[req.VolumeId]; ok {
+		volInfo.ISPublished = true
```
Comment (jsafrane) [inline mock/service/controller.go:231]: `ISPublished` is used for NodePublish/Unpublish, not for ControllerPublish. If you need to check for attached volumes before resizing them, add a separate IsControllerPublished flag to `volInfo`.

Which brings me to question, expanding an attached and unmounted volume is online or offline resize?

Comment (gnufied) [reply mock/service/controller.go:231]: it is still considered online.

Comment (gnufied) [reply mock/service/controller.go:231]: ack.  the offline check does uses `IsPublished` check, so I will update that code too.

**Pull request merged**